### ISSUE TITLE
[Bug] deserialization from value into Option fails

### ISSUE THREAD

**Issue by CR7273868** [belak/serde-xmlrpc#22]
Given code:

#[derive(Deserialize, Serialize)]
pub struct LoginToReturnFromXMLRPC {
    #[serde(rename = "avatarName")]
    username: String,
    pub email: String,
    #[serde(rename = "userId")]
    pub user_id: u32,
    #[serde(rename = "securityKey")]
    pub security_key: String,
    // Skip deserializing this field if it's None
    pub device_login_token: Option<String>,
}

See device_login_token. Might be nil might not be. That's why I gave it an Option<String>. if it's some then the decoder in serdexml_rpc will give out the following error.

Error: Bug:  serde: invalid type: string \"3026b5dafa5aae3210096e30649ff31c\", expected option.

So the library basically doesn't turn it into an Option<String> if it's already a string.

**Comment by belak** [comment]
I can reproduce this with a minimal example:

```
use serde::{Serialize, Deserialize};

#[derive(Deserialize, Serialize, Debug)]
pub struct Sample {
    token: Option<String>,
}

fn main() {
    println!("json: {:?}", serde_json::from_str::<Sample>(r#"{
  "token": "3026b5dafa5aae3210096e30649ff31c"
}"#));

    println!("xmlrpc: {:?}", serde_xmlrpc::response_from_str::<Sample>(r###"<?xml version="1.0" encoding="utf-8"?>
  <methodResponse>
    <params>
      <param>
        <value>
          <struct>
            <member>
              <name>token</name>
              <value><string>3026b5dafa5aae3210096e30649ff31c</string></value>
            </member>
          </struct>
        </value>
      </param>
    </params>
  </methodResponse>"###));
}
```

This outputs:

```
json: Ok(Sample { token: Some("3026b5dafa5aae3210096e30649ff31c") })
xmlrpc: Err(DecodingError(SerdeError("invalid type: string \"3026b5dafa5aae3210096e30649ff31c\", expected option")))
```

That shows the json crate handles it correctly, while xmlrpc doesn't. I'm not entirely sure why - I took a quick look this morning and didn't find anything obvious. That being said I am not a serde expert by any means, and I don't use xmlrpc regularly. I'll try to take a look, but PRs for this would be appreciated.

Note that the design was based on the official serde-json package, hence why I'm using that for a reference in the sample.

**Comment by CR7273868** [comment]
I'll try to help u on this and if I have a solution I'll PR. 

I was looking into the library and true, it's a very weird bug I didn't find anything obvious either. Any attempt to fix or look thru references a lot all ended up into a dead end.

**Comment by belak** [comment]
I believe I've found it - the Deserializer in src/util/value.rs is missing a separate `deserialize_option` method and requires some minor refactoring to get it to work.

**Comment by CR7273868** [comment]
Oh that went completely over my head.

**Comment by belak** [comment]
This should be resolved by commit 30778eb169f19a4a736b8a1783327d6c2356605e and has been pushed to crates.io as v0.2.1.

Thanks for reporting this! Good catch.
